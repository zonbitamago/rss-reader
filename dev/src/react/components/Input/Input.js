import React, { Component } from "react";
import PropTypes from "prop-types";
import styles from "./Input.css";
import TextField from "@material-ui/core/TextField";

class Input extends Component {
  constructor(props) {
    super(props);
    this.state = { val: this.props.value != undefined ? this.props.value : "" };
    this.changeVal = this.changeVal.bind(this);
  }

  changeVal(e) {
    this.setState({ val: e.target.value });
    if (this.props.changeParentVal != undefined) {
      this.props.changeParentVal(e.target.value);
    }
  }

  render() {
    return (
      <div className="Input">
        <TextField
          required
          id={this.props.name}
          label={this.props.name}
          onChange={this.changeVal}
          value={this.state.val}
        />
      </div>
    );
  }
}

Input.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.any,
  changeParentVal: PropTypes.func
};

Input.defaultProps = {};

export default Input;
