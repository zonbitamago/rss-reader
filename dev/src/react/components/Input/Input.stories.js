import React from "react";
import { storiesOf } from "@storybook/react";

import Input from "./Input";

storiesOf("Atoms/Input", module)
  .add("URL", () => <Input name="URL" />)
  .add("Name", () => <Input name="Name" />)
  .add("UpdateDuration", () => <Input name="UpdateDuration" value="5" />);
