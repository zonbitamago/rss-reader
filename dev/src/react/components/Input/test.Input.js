import React from "react";
import { shallow } from "enzyme";

import Input from "./Input";

describe("Input", () => {
  let component, props;

  beforeEach(() => {
    // props = { name: "test",changeParentVal:(val) => {} };
    // component = shallow(<Input {...props} />);
  });

  // it('should', () => {
  //   expect(component).toMatchSnapshot()
  // })
  describe("フィールド入力確認", () => {
    it("呼び出し元変数変更　あり", () => {
      props = { name: "test", changeParentVal: val => {} };
      component = shallow(<Input {...props} />);
      component.find("#test").simulate("change", { target: { value: "abc" } });
      expect(component.state("val")).toBe("abc");
    });

    it("呼び出し元変数変更　なし", () => {
      props = { name: "test" };
      component = shallow(<Input {...props} />);
      component.find("#test").simulate("change", { target: { value: "abc" } });
      expect(component.state("val")).toBe("abc");
    });
  });

  it("初期値入力済み確認", () => {
    props = { name: "test", value: "" };
    component = shallow(<Input {...props} />);
    expect(component).toMatchSnapshot();
  });
});
