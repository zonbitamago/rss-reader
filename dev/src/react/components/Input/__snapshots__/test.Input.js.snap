// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input 初期値入力済み確認 1`] = `
<div
  className="Input"
>
  <TextField
    id="test"
    label="test"
    onChange={[Function]}
    required={true}
    select={false}
    value=""
  />
</div>
`;
